tests/services/analysis_starter/test_balsamic_tracker.py

```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+from unittest.mock import Mock, create_autospec
+
+import pytest
+
+from cg.apps.tb import TrailblazerAPI
+from cg.apps.tb.models import TrailblazerAnalysis
+from cg.constants import Workflow
+from cg.services.analysis_starter.configurator.models.balsamic import BalsamicCaseConfig
+from cg.services.analysis_starter.tracker.implementations.balsamic import BalsamicTracker
+from cg.store.models import Analysis
+from cg.store.store import Store
+
+
+@pytest.fixture
+def balsamic_tracker(balsamic_store: Store, tmp_path: Path) -> BalsamicTracker:
+    balsamic_root = tmp_path / "balsamic"
+    balsamic_root.mkdir(parents=True, exist_ok=True)
+    trailblazer_api: Mock[TrailblazerAPI] = create_autospec(TrailblazerAPI)
+    return BalsamicTracker(
+        store=balsamic_store,
+        trailblazer_api=trailblazer_api,
+        workflow_root=str(balsamic_root),
+    )
+
+
+def test_balsamic_tracker_successful(
+    balsamic_case_config: BalsamicCaseConfig,
+    balsamic_tracker: BalsamicTracker,
+):
+    # GIVEN a valid Balsamic case config
+
+    # GIVEN that a Balsamic Config file exists
+    balsamic_case_config.sample_config.parent.mkdir()
+    with open(balsamic_case_config.sample_config, "w") as file:
+        file.write('{"analysis": {"BALSAMIC_version": "1.0.12"}}')
+
+    balsamic_tracker.trailblazer_api.add_pending_analysis = Mock(
+        return_value=create_autospec(TrailblazerAnalysis, id=11234)
+    )
+    balsamic_tracker.track(case_config=balsamic_case_config)
+    # WHEN tracking the analysis
+
+    # THEN trailblazer API should be called with the correct parameters
+    balsamic_tracker.trailblazer_api.add_pending_analysis.assert_called_once()
```

Comment:
I think this `assert_called_once()` leaves some of the functions in this class untested. I like the approach taken when testing the microSALT tracker here :https://github.com/Clinical-Genomics/cg/blob/9b392d08282c8a5a0ffe46b3d23e6d86b89533a0/tests/services/analysis_starter/test_microsalt_tracker.py#L60